irods/test/data_obj_test.py

```diff
@@ -1860,6 +1860,41 @@ def test_access_through_resc_hierarchy__243(self):
         finally:
             s.resources.remove('parent')
 
+
+    def test_temporary_xml_mode_change_with_operation_as_proof__issue_586(self):
+        from irods.message import (XML_Parser_Type, current_XML_parser, string_for_XML_parser)
+        active_xml_for_thread = lambda : string_for_XML_parser(current_XML_parser())
+        from irods.helpers import xml_mode
+        sess = irods.test.helpers.make_session()
+        hc = irods.test.helpers.home_collection(sess)
+        name = '{hc}/\1'.format(**locals())
+
+        self.assertEqual(active_xml_for_thread(), 'STANDARD_XML')
+
+        with xml_mode('QUASI_XML'):
+            sess.data_objects.create(name)
+
+        self.assertEqual(active_xml_for_thread(), 'STANDARD_XML')
+        try:
+            with self.assertRaises(xml.etree.ElementTree.ParseError):
+                sess.collections.get(hc).data_objects
+        finally:
+            with xml_mode('QUASI_XML'):
+                sess.data_objects.unlink(name, force = True)
```

Comment:
Are `try` and `finally` actually needed here?

Reply:
Only if something goes wrong with sess.collections.get ... which it won't. Will remove `try/finally`.

Reply:
Had a change of heart, leaving try/finally in , I have decided.  If, in the future, the  assertion of which or whether exceptions are raised is foiled, in any case I'd rather not leave the data object around.

Reply:
Fine with me.